extras/mosquitto/Dockerfile

```diff
@@ -0,0 +1,95 @@
+    rm /tmp/lws.tar.gz && \
+    cd /build/lws && \
+    cmake . \
+        -DCMAKE_BUILD_TYPE=MinSizeRel \
+        -DCMAKE_INSTALL_PREFIX=/usr \
+        -DLWS_IPV6=ON \
+        -DLWS_WITHOUT_BUILTIN_GETIFADDRS=ON \
+        -DLWS_WITHOUT_CLIENT=ON \
+        -DLWS_WITHOUT_EXTENSIONS=ON \
+        -DLWS_WITHOUT_TESTAPPS=ON \
+        -DLWS_WITH_SHARED=OFF \
+        -DLWS_WITH_ZIP_FOPS=OFF \
+        -DLWS_WITH_ZLIB=OFF && \
+    make -j "$(nproc)" && \
+    rm -rf /root/.cmake && \
+    wget https://mosquitto.org/files/source/mosquitto-${VERSION}.tar.gz -O /tmp/mosq.tar.gz && \
+    echo "$DOWNLOAD_SHA256  /tmp/mosq.tar.gz" | sha256sum -c - && \
+    wget https://mosquitto.org/files/source/mosquitto-${VERSION}.tar.gz.asc -O /tmp/mosq.tar.gz.asc && \
+    export GNUPGHOME="$(mktemp -d)" && \
+    found=''; \
+    for server in \
+        ha.pool.sks-keyservers.net \
+        hkp://keyserver.ubuntu.com:80 \
+        hkp://p80.pool.sks-keyservers.net:80 \
+        pgp.mit.edu \
+    ; do \
+        echo "Fetching GPG key $GPG_KEYS from $server"; \
+        gpg --keyserver "$server" --keyserver-options timeout=10 --recv-keys "$GPG_KEYS" && found=yes && break; \
+    done; \
+    test -z "$found" && echo >&2 "error: failed to fetch GPG key $GPG_KEYS" && exit 1; \
+    gpg --batch --verify /tmp/mosq.tar.gz.asc /tmp/mosq.tar.gz && \
+    gpgconf --kill all && \
+    rm -rf "$GNUPGHOME" /tmp/mosq.tar.gz.asc && \
+    mkdir -p /build/mosq && \
+    tar --strip=1 -xf /tmp/mosq.tar.gz -C /build/mosq && \
+    rm /tmp/mosq.tar.gz && \
+    make -C /build/mosq -j "$(nproc)" \
+        CFLAGS="-Wall -O2 -I/build/lws/include -flto" \
+        LDFLAGS="-L/build/lws/lib -flto" \
+        WITH_ADNS=no \
+        WITH_DOCS=no \
+        WITH_MEMORY_TRACKING=no \
+        WITH_SHARED_LIBRARIES=no \
+        WITH_SRV=no \
+        WITH_STRIP=yes \
+        WITH_TLS_PSK=no \
+        WITH_WEBSOCKETS=yes \
+        prefix=/usr \
+        binary && \
+    addgroup -S -g 1883 mosquitto 2>/dev/null && \
+    adduser -S -u 1883 -D -H -h /var/empty -s /sbin/nologin -G mosquitto -g mosquitto mosquitto 2>/dev/null && \
+    mkdir -p /mosquitto/config /mosquitto/data /mosquitto/log && \
+    install -d /usr/sbin/ && \
+    install -s -m755 /build/mosq/src/mosquitto /usr/sbin/mosquitto && \
+    install -s -m755 /build/mosq/src/mosquitto_passwd /usr/bin/mosquitto_passwd && \
+    install -m644 /build/mosq/mosquitto.conf /mosquitto/config/mosquitto.conf && \
+    chown -R mosquitto:mosquitto /mosquitto && \
+    apk --no-cache add \
+        libuuid && \
+    apk del build-deps && \
+    rm -rf /build
+
+# MQTT-CARBON PROXY
+RUN apk add --no-cache python3
+RUN pip3 install --no-cache-dir paho-mqtt
+VOLUME ["/mosquitto/data", "/mosquitto/log"]
+
+RUN apk add supervisor
+RUN mkdir -p /var/log/supervisor
+COPY extras/mosquitto/supervisord.conf /etc/supervisord.conf
+
+COPY extras/mosquitto/mqtt-carbon-proxy.py /
+
+CMD ["/usr/bin/supervisord", "-c", "/etc/supervisord.conf"]
+
```

Comment:
Why copying the Dockerfile from someone else rather than using it as base?

Reply:
yes, we could do it like that

Reply:
I've just opened an issue with that. I try to keep all the images from the same alpine version, this is why I used the author original Dockerfile and modified. But we could discuss that.